bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,823 @@
+        }
+
+        // Scan finished but error. clear the list
+        if (ret == false) {
+            listOfComponentId.clear();
+        }
+
+        return listOfComponentId;
+    }
+
+    /**
+     * Method to send http request to get the device description
+     */
+    public FreeAtHomeDeviceDescription getFreeatHomeDeviceDescription(String id)
+            throws FreeAtHomeHttpCommunicationException {
+        FreeAtHomeDeviceDescription device = new FreeAtHomeDeviceDescription();
+
+        String url = baseUrl + "/rest/device/" + sysApUID + "/" + id;
+        try {
+            HttpClient client = httpClient;
+            Request req = client.newRequest(url);
+
+            if (req == null) {
+                throw new FreeAtHomeHttpCommunicationException(0,
+                        "Invalid request object in getDatapoint with the URL [ " + url + " ]");
+            }
+
+            ContentResponse response;
+            response = req.send();
+
+            // Get component List
+            String deviceString = new String(response.getContent());
+
+            JsonReader reader = new JsonReader(new StringReader(deviceString));
+            reader.setLenient(true);
+            JsonElement jsonTree = JsonParser.parseReader(reader);
+
+            if (!jsonTree.isJsonObject()) {
+                throw new FreeAtHomeHttpCommunicationException(0,
+                        "No data is received by getDatapoint with the URL [ " + url + " ]");
+            }
+
+            if (!jsonTree.isJsonObject()) {
+                throw new FreeAtHomeHttpCommunicationException(0,
+                        "Invalid jsonObject in getFreeatHomeDeviceDescription with the URL [ " + url + " ]");
+            }
+
+            // check the output
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+
+            if (!jsonObject.isJsonObject()) {
+                throw new FreeAtHomeHttpCommunicationException(0,
+                        "Main jsonObject is invalid in getFreeatHomeDeviceDescription with the URL [ " + url + " ]");
+            }
+
+            jsonObject = jsonObject.getAsJsonObject(sysApUID);
+
+            if (!jsonObject.isJsonObject()) {
+                throw new FreeAtHomeHttpCommunicationException(0,
+                        "jsonObject is invalid in getFreeatHomeDeviceDescription with the URL [ " + url + " ]");
+            }
+
+            jsonObject = jsonObject.getAsJsonObject("devices");
+
+            if (!jsonObject.isJsonObject()) {
+                throw new FreeAtHomeHttpCommunicationException(0,
+                        "Devices Section is missing in getFreeatHomeDeviceDescription with the URL [ " + url + " ]");
+            }
+
+            device = new FreeAtHomeDeviceDescription(jsonObject, id);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            logger.debug("No communication possible to get device list - Communication interrupt [ {} ]",
```

Comment:
For `InterruptedException`, restore the interrupted state of the thread by calling `Thread.currentThread().interrupt()`, see https://rules.sonarsource.com/java/RSPEC-2142/